electrum/simple_config.py

```diff
@@ -283,6 +285,10 @@ def get(self, key: str, default=None) -> Any:
             out = self.cmdline_options.get(key)
             if out is None:
                 out = self.user_config.get(key, default)
+                if out is None:
+                    cv = config_vars.get(key)
+                    if cv:
+                        out = cv._get_config_value(self)
```

Comment:
Maybe we could move this (or similar) logic to the CLI getconfig/setconfig commands.

And also we could change the name of the `config.get` and `config.set_key` methods to start with an underscore. These are low level methods that should typically not be called from outside this file.

Reply:
We could do something like this:
```patch
diff --git a/electrum/commands.py b/electrum/commands.py
index 6be08777f2..befab360ee 100644
--- a/electrum/commands.py
+++ b/electrum/commands.py
@@ -305,7 +305,8 @@ class Commands:
     @command('')
     async def getconfig(self, key):
         """Return a configuration variable. """
-        return self.config.get(key)
+        cv = self.config.cv.from_key(key)
+        return cv.get()
 
     @classmethod
     def _setconfig_normalize_value(cls, key, value):
@@ -326,7 +327,8 @@ class Commands:
             self.daemon.commands_server.rpc_user = value
         if self.daemon and key == SimpleConfig.RPC_PASSWORD.key():
             self.daemon.commands_server.rpc_password = value
-        self.config.set_key(key, value)
+        cv = self.config.cv.from_key(key)
+        cv.set(value)
         return True
 
     @command('')
diff --git a/electrum/simple_config.py b/electrum/simple_config.py
index 38d70c2a71..9637417cc0 100644
--- a/electrum/simple_config.py
+++ b/electrum/simple_config.py
@@ -52,6 +52,9 @@ _logger = get_logger(__name__)
 FINAL_CONFIG_VERSION = 3
 
 
+_config_var_from_key = {}  # type: Dict[str, 'ConfigVar']
+
+
 class ConfigVar(property):
 
     def __init__(
@@ -65,6 +68,7 @@ class ConfigVar(property):
         self._default = default
         self._type = type_
         property.__init__(self, self._get_config_value, self._set_config_value)
+        _config_var_from_key[key] = self
 
     def _get_config_value(self, config: 'SimpleConfig'):
         with config.lock:
@@ -818,10 +822,18 @@ class SimpleConfig(Logger):
         """
         class CVLookupHelper:
             def __getattribute__(self, name: str) -> ConfigVarWithConfig:
+                if name in ("from_key", ):  # don't apply magic, just use standard lookup
+                    return super().__getattribute__(name)
                 config_var = config.__class__.__getattribute__(type(config), name)
                 if not isinstance(config_var, ConfigVar):
                     raise AttributeError()
                 return ConfigVarWithConfig(config=config, config_var=config_var)
+            def from_key(self, key: str) -> ConfigVarWithConfig:
+                try:
+                    config_var = _config_var_from_key[key]
+                except KeyError:
+                    raise Exception(f"No ConfigVar with key={key!r}") from None
+                return ConfigVarWithConfig(config=config, config_var=config_var)
             def __setattr__(self, name, value):
                 raise Exception(
                     f"Cannot assign value to config.cv.{name} directly. "

```